src/evm/stream.rs

```diff
@@ -18,6 +18,34 @@ use crate::{
     },
 };
 
+/// Builds the protocol stream, providing a `BlockUpdate` for each block received.
+///
+/// Each `BlockUpdate` can then be used at a higher level to retrieve important information from
+/// the block, such as the updated states of protocol components, which can in turn be used
+/// to obtain spot price information for a desired component and token pair.
+///
+/// # Important
+/// Decoding is performed using the [`TychoStreamDecoder`](crate::evm::decoder::TychoStreamDecoder).
```

Comment:
I just realised that this class is private, so we can't link to it's docs :/ So let's move the corresponding docs from there to this struct, I think it is better to have it all here.